lib/layer.js

```diff
@@ -41,10 +41,53 @@ function Layer (path, options, fn) {
   this.name = fn.name || '<anonymous>'
   this.params = undefined
   this.path = undefined
-  this.regexp = pathRegexp((opts.strict ? path : loosen(path)), this.keys, opts)
+  this.slash = path === '/' && opts.end === false
+
+  function matcher (_path) {
+    if (_path instanceof RegExp) {
+      var keys = []
+      var name = 0
+      var m
+      // eslint-disable-next-line no-cond-assign
+      while (m = MATCHING_GROUP_REGEXP.exec(_path.source)) {
+        keys.push({
+          name: m[1] || name++,
+          offset: m.index
+        })
+      }
+
+      return function regexpMatcher (p) {
+        var match = _path.exec(p)
+        if (!match) {
+          return false
+        }
+
+        var params = {}
+        for (var i = 1; i < match.length; i++) {
+          var key = keys[i - 1]
+          var prop = key.name
+          var val = decodeParam(match[i])
+
+          if (val !== undefined || !(hasOwnProperty.call(params, prop))) {
```

Comment:
I think you can get rid of the `hasOwnProperty`? Last write wins has been how it worked forever already.

Reply:
ah yeah, that was just copy pasta.